### PR TITLE
Fixed #25987 -- Made inline formset validation respect unique_together with an unsaved parent object.

### PR DESCRIPTION
**Pull request by charettes**
Thanks Anton Kuzmichev for the report.

### CODE REVIEW

File: django/forms/models.py [hunk 1]
```diff
@@ -564,6 +564,9 @@ class BaseModelFormSet(BaseFormSet):
     """
     model = None
 
+    # Set of fields assumed to be unique among forms of this set.
```
Comment (timgraham) [inline django/forms/models.py]: assumed to be -> that must be?

File: django/forms/models.py [hunk 2]
```diff
@@ -677,9 +680,12 @@ def validate_unique(self):
         for uclass, unique_check in all_unique_checks:
             seen_data = set()
             for form in valid_forms:
-                # get data for each field of each of unique_check
-                row_data = (form.cleaned_data[field]
-                            for field in unique_check if field in form.cleaned_data)
+                # Get data for each field of each of unique_check and names
```
Comment (timgraham) [inline django/forms/models.py]: # Get the data for the set of fields that must be unique among the forms.

File: tests/inline_formsets/tests.py [hunk 3]
```diff
@@ -162,3 +162,17 @@ def test_zero_primary_key(self):
...
+        data = {
+            'poem_set-TOTAL_FORMS': '2',
+            'poem_set-INITIAL_FORMS': '0',
+            'poem_set-MAX_NUM_FORMS': '2',
+            'poem_set-0-name': 'Poem',
+            'poem_set-1-name': 'Poem',
+        }
+        formset = PoemFormSet(data, instance=poet)
+        self.assertFalse(formset.is_valid())
+        self.assertEqual(formset.errors, [{}, {'__all__': [formset.get_form_error()]}])
```
Comment (timgraham) [inline tests/inline_formsets/tests.py]: I'm not sure why `get_form_error()` is named as it is. I would find the test more readable if you replaced the method call with the "Please correct the duplicate values below." string, but whatever you think.

Comment (timgraham) [inline tests/inline_formsets/tests.py]: Alternatively, it could help to assert `formset[0].errors` -> `['Please correct the duplicate data for name.']`.

Comment (charettes) [inline tests/inline_formsets/tests.py]: Yeah I think it makes more sense to assert for `formset[0].errors` instead.

**Comment by timgraham** [comment]
Looks good, I tried to suggest a slightly more descriptive commit message ("adjusted" is a bit vague).

**Pull request merged**